android/src/main/java/com/swmansion/rnscreens/Screen.kt

```diff
@@ -236,6 +242,34 @@ class Screen constructor(context: ReactContext?) : FabricEnabledViewGroup(contex
             mNativeBackButtonDismissalEnabled = enableNativeBackButtonDismissal
         }
 
+    fun recalculateHeaderHeight() {
+        val summarizedHeight = when (stackPresentation) {
+            StackPresentation.TRANSPARENT_MODAL -> 0
+            else -> getSummarizedHeaderHeight()
+        }
+
+        UIManagerHelper.getEventDispatcherForReactTag(context as ReactContext, id)
+            ?.dispatchEvent(HeaderHeightChangeEvent(id, summarizedHeight))
+    }
+
+    private fun getSummarizedHeaderHeight(): Int {
```

Comment:
👍 to what's @kkafar saying about naming and let's also try to use similar naming on both platforms